backends/ebpf/psa/ebpfPsaArch.cpp

```diff
@@ -0,0 +1,637 @@
+
+    parser->visitor->useAsPointerVariable(resubmit_meta->name.name);
+    parser->visitor->useAsPointerVariable(parser->user_metadata->name.name);
+    parser->visitor->useAsPointerVariable(parser->headers->name.name);
+
+    return true;
+}
+
+// =====================EBPFControl=============================
+bool ConvertToEBPFControlPSA::preorder(const IR::ControlBlock *ctrl) {
+    control = new EBPFControlPSA(program,
+                                 ctrl,
+                                 parserHeaders);
+    program->control = control;
+    program->to<EBPFPipeline>()->control = control;
+    control->hitVariable = refmap->newName("hit");
+    auto pl = ctrl->container->type->applyParams;
+    auto it = pl->parameters.begin();
+    control->headers = *it; ++it;
+    control->user_metadata = *it; ++it;
+    control->inputStandardMetadata = *it; ++it;
+    control->outputStandardMetadata = *it;
+
+    auto codegen = new ControlBodyTranslator(control);
+    codegen->substitute(control->headers, parserHeaders);
+
+    if (type != TC_EGRESS) {
+        codegen->useAsPointerVariable(control->outputStandardMetadata->name.name);
+    }
+
+    codegen->useAsPointerVariable(control->headers->name.name);
+    codegen->useAsPointerVariable(control->user_metadata->name.name);
+
+    control->codeGen = codegen;
+
+    for (auto a : ctrl->constantValue) {
+        auto b = a.second;
+        if (b->is<IR::Block>()) {
+            this->visit(b->to<IR::Block>());
+        }
+    }
+    return true;
+}
+
+bool ConvertToEBPFControlPSA::preorder(const IR::TableBlock *tblblk) {
+    // use HASH_MAP as default type
+    TableKind tableKind = TableHash;
+
+    // If any key field is LPM we will generate an LPM table
+    auto keyGenerator = tblblk->container->getKey();
+    if (keyGenerator != nullptr) {
+        for (auto it : keyGenerator->keyElements) {
+            // optimization: check if we should generate timestamp
+            if (it->expression->toString().endsWith("timestamp")) {
+                control->timestampIsUsed = true;
+            }
+
+            auto mtdecl = refmap->getDeclaration(it->matchType->path, true);
+            auto matchType = mtdecl->getNode()->to<IR::Declaration_ID>();
+            if (matchType->name.name != P4::P4CoreLibrary::instance.exactMatch.name &&
+                matchType->name.name != P4::P4CoreLibrary::instance.lpmMatch.name)
+                ::error(ErrorType::ERR_UNSUPPORTED,
+                        "Match of type %1% not supported", it->matchType);
+
+            if (matchType->name.name == P4::P4CoreLibrary::instance.lpmMatch.name) {
+                if (tableKind == TableLPMTrie) {
+                    ::error(ErrorType::ERR_UNSUPPORTED,
+                            "%1%: only one LPM field allowed", it->matchType);
+                    return false;
+                }
+                tableKind = TableLPMTrie;
+            }
+        }
+    }
+
+    // use 1024 by default
+    size_t size = 1024;
```

Comment:
hidden like this it will be hard to find and change. Perhaps it should also be a compiler option.